### PR TITLE
Code improvements

### PR DESCRIPTION
**Pull request by othalla**

### CODE REVIEW

File: celery/canvas.py [hunk 1]
```diff
@@ -1380,7 +1380,7 @@ def _get_app(self, body=None):
                 tasks = self.tasks.tasks  # is a group
             except AttributeError:
                 tasks = self.tasks
-            if tasks:
+            if len(tasks):
```
Comment (auvipy) [inline celery/canvas.py]: the reason for this change?

Comment (othalla) [reply celery/canvas.py]: using len is quite heavy to know if a list or something is empty as it requires parsing it totaly

Comment (clokep) [reply celery/canvas.py]: In Python, calling `len()` on a `list` (or `dict` or `set`) is `O(1)` (see this [StackOverflow](https://stackoverflow.com/questions/12342457/what-is-the-big-o-notation-for-the-len-function-in-python) for more information). It *does not* require iterating the list.

Personally, I find it more readable without the `len()` though. I don't think there's a performance benefit or penalty either way.

Comment (othalla) [reply celery/canvas.py]: okok i see

**Comment by thedrow** [comment]
What's the status here?

**Comment by othalla** [comment]
@thedrow i don't have any news about this PR, should i fix something?

**Comment by thedrow** [comment]
Simply rebase. I need to see CI green before merging this.

**Comment by othalla** [comment]
@thedrow  done rebase

**Comment by clokep** [comment]
I think this had unhandled comments above about the use of `len()`?

**Comment by thedrow** [comment]
Yeh we can remove that one...

**Comment by othalla** [comment]
should i do something regarding your comments ?

**Comment by thedrow** [comment]
Just remove len from that condition.

**Pull request merged**